### PR TITLE
realm roles: associated roles modal (roles list)

### PR DESCRIPTION
**Pull request by jenny-s51**
## Motivation

#284 , https://issues.redhat.com/browse/APPDUX-746

## Brief Description

Builds out modal for adding associated roles.

## Verification Steps

1. Go to Realm Roles section.
2. Click on one of the roles in the list.
3. Go to the Actions dropdown in the upper right.
4. Click "Add associated roles"
5. Verify that the modal renders and displays the list of existing realm roles. 

## Checklist:

- [x] Code has been tested locally by PR requester
- [ ] User-visible strings are using the react-i18next framework (useTranslation)
- [ ] Help has been implemented
- [ ] axe report has been run and resulting a11y issues have been resolved
- [ ] Unit tests have been created/updated
- [x] Formatting has been performed via prettier/eslint
- [x] Type checking has been performed via 'yarn check-types'

### CODE REVIEW

**Comment by edewit** [comment]
I guess the list of this dialog it should not have the role itself. I mean, if I want to associate the role "admin" with another role the role "admin" should not be in the dialog?

**Comment by jenny-s51** [comment]
@edewit nice catch - fixed!

File: src/realm-roles/RealmRolesSection.tsx [hunk 1]
```diff
@@ -71,6 +71,26 @@ export const RealmRolesSection = () => {
...
+  //       await adminClient.roles.delById({
+  //         id: selectedRole!.id!,
+  //       });
+  //       setSelectedRole(undefined);
+  //       addAlert(t("roleDeletedSuccess"), AlertVariant.success);
+  //     } catch (error) {
+  //       addAlert(`${t("roleDeleteError")} ${error}`, AlertVariant.danger);
+  //     }
+  //   },
+  // });
```
Comment (edewit) [inline src/realm-roles/RealmRolesSection.tsx:92]: Let's remove these comments
```suggestion
```

File: src/realm-roles/AssociatedRolesModal.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    });
+  };
+
+  const boolFormatter = (): IFormatter => (data?: IFormatterValueType) => {
+    const boolVal = data?.toString();
+
+    return (boolVal
+      ? boolVal.charAt(0).toUpperCase() + boolVal.slice(1)
+      : undefined) as string;
+  };
```
Comment (edewit) [inline src/realm-roles/AssociatedRolesModal.tsx:76]: we could move this to `./util.js`, it's used in 2 places I think

**Review by edewit — CHANGES_REQUESTED** [review]

File: src/realm-roles/AssociatedRolesModal.tsx (hunk 2 above)
Comment (jenny-s51) [reply src/realm-roles/AssociatedRolesModal.tsx]: great idea - done

File: src/realm-roles/messages.json [hunk 3]
```diff
@@ -1,8 +1,12 @@
 {
   "roles": {
     "attributes": "Attributes",
+    "add": "Add",
```
Comment (mfrances17) [inline src/realm-roles/messages.json:4]: You should use the Add string that already exists in common

**Review by mfrances17 — CHANGES_REQUESTED** [review]
Getting the following error when selecting the Action menu or one of its commands:
`Encountered two children with the same key, `.$action`. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version.`
Your menu commands are probably using the same key.

Also, if I add associated roles, what is the expected result? I don't see any way to see associated roles to verify that they've been added.

File: src/realm-roles/messages.json (hunk 3 above)
Comment (jenny-s51) [reply src/realm-roles/messages.json]: @mfrances17 nice catch - updated!

**Comment by jenny-s51** [comment]
@mfrances17 thank you for your helpful feedback! I fixed the duplicate keys and the "add"/"cancel" strings.

This PR is meant to address just the implementation of a selectable modal populated with realm roles from the backend. I plan on addressing the functionality in a separate PR (see JIRA issue: https://issues.redhat.com/browse/APPDUX-791) and  https://github.com/jenny-s51/keycloak-admin/tree/associated-roles-modal-functionality

**Review by edewit — APPROVED** [review]

**Comment by ssilvert** [comment]
@mfrances17 Are you ok with merging this?

**Review by mfrances17 — APPROVED** [review]

**Comment by mfrances17** [comment]
Yep, with Jenny's explanation that this isn't meant for that functionality yet, this can be merged. Need to resolve conflicts first, of course.

**Pull request merged**